trio/_highlevel_open_tcp_stream.py

```diff
@@ -283,7 +301,11 @@ async def open_tcp_stream(
     #   the next connection attempt to start early
     # code needs to ensure sockets can be closed appropriately in the
     # face of crash or cancellation
-    async def attempt_connect(socket_args, sockaddr, attempt_failed):
+    async def attempt_connect(
+        socket_args: tuple[AddressFamily, SocketKind],
+        sockaddr: tuple[str, int] | tuple[str, int, int, int],
```

Comment:
Use `Address` ?

Reply:
`attempt_connect` is called from unpacking the results of `getaddrinfo`, and with that unpacking, `sockaddr` can only become `tuple[str, int] | tuple[str, int, int, int]` unless the return annotation for `getaddrinfo` is wrong. I do see that it uses `sockaddr` for `socket.connect`, and `socket.connect`'s annotation says it accepts the `Address` union, but I feel like annotating `attempt_connect`'s `sockaddr` argument to be `Address` wouldn't be telling the truth about what we expect `getaddrinfo` to tell us.

Reply:
Informing what `getaddrinfo` gives should be up to `getaddrinfo`, no? `attempt_connect`s signatures responsibility is saying what it accepts and returns, and it can accept `Address` no problem. Though it's a nested function it barely matters, other than if `open_tcp_stream` was changed to modify the address it got from `getaddrinfo` before sending it to `attempt_connect` - but anybody attempting such a change shouldn't be hindered by a restricting type hint on it's helper functions IMO.

Reply:
Changed in e82ce69

Reply:
Basically, accepting this causes mypy errors, and so this change was reverted in afaf153